source/simulator/core.cc

```diff
@@ -1037,14 +1080,29 @@ namespace aspect
 
     Table<2,DoFTools::Coupling> coupling (introspection.n_components,
                                           introspection.n_components);
-    for (unsigned int c=0; c<introspection.n_components; ++c)
-      for (unsigned int d=0; d<introspection.n_components; ++d)
-        if (c == d)
-          coupling[c][d] = DoFTools::always;
-        else
-          coupling[c][d] = DoFTools::none;
+
+    const typename Introspection<dim>::ComponentIndices &x
+      = introspection.component_indices;
+
+    for (unsigned int c=0; c<dim; ++c)
```

Comment:
It seems previously we had `c` and `d` indices, but now that we have only one, `d` would be the appropriate one for a loop over the dimension. Nitpicky me :smile: